src/utils.js

```diff
@@ -47,29 +58,39 @@ export const createHistogramArray = (dist) => {
         bin_indices);
 };
 
-export const getDomain = (hist) => {
-    let domain = hist.map((e) => e[0]);
-    return [
-        Math.min(...domain),
-        Math.max(...domain)];
-};
+export const createScatterPlotHistogram = (samples, bins, minum, maxum) => {
+    /* @samples An array of values to plot
+     * @bins    The number of bins
+     * @minum   The smallest number that the bins should align to
+     * @maxum   The largest number that bins should align to */
 
-export const getHistogramMaxima = (hist) => {
-    return Math.max(...hist.map((e) => e[1]));
-};
+    // Creates an 'ordered' histogram, where each value is
+    // represented in the histogram rather than accumulated;
+    // and where each value is inserted in the order of the passed array
+    let nBins = bins || NO_OF_BINS;
+    let min = minum;
+    let max = maxum;
+    let bin_size = (max - min) / nBins;
 
-export const interpolateHistogram = (hist) => {
-    // The function fills in data with values less than the
-    // original histogram value.
-    return hist.reduce((acc, e) => {
-        // [val, int]
-        acc.push(e);
-        if (e[1] > 1) {
-            math.range(1, e[1]).map((i) => {
-                acc.push([e[0], i]);
-            });
-        }
+    let flatBinIndicies = getBinIndicies(nBins, bin_size, min).map((e) => e[0]);
+    let binned_values = getBinnedValues(samples, nBins, bin_size, min);
+
+    return binned_values.reduce((acc, val) => {
+        // For each binned_value find the greatest frequency in the accumulator
+        // up to that point.
+        let maxFreq = 0;
+        let bin = flatBinIndicies[val];
+        acc.map((v) => {
+            if (v[0] == bin) {
```

Comment:
Use triple equals for consistency here. It's just one less thing to think about.. unless there's a weird reason for using the more lax `==`.

Reply:
Nope, no reason - good point.